code/modules/overmap/exoplanets/turfs.dm

```diff
@@ -146,14 +145,16 @@
 		var/obj/effect/overmap/visitable/sector/exoplanet/E = map_sectors["[z]"]
 		if(istype(E) && E.grass_color)
 			color = E.grass_color
-	if(!resources)
-		resources = list()
+
+	var/datum/extension/buried_resources/resources = get_or_create_extension(src, /datum/extension/buried_resources)
 	if(prob(70))
-		resources[/decl/material/solid/mineral/graphite] = rand(3,5)
+		LAZYSET(resources.resources, /decl/material/solid/mineral/graphite, rand(3,5))
 	if(prob(5))
-		resources[/decl/material/solid/metal/uranium] = rand(1,3)
+		LAZYSET(resources.resources, /decl/material/solid/metal/uranium, rand(1,3))
```

Comment:
Responsibilities are a bit vague here, with a user of the extension having to know that the `resources` list is a lazy-list
Preferable would probably be to have an `/AddResource(type, amount)` proc on the extension (and also mark the `resources` list as `VAR_PRIVATE`/`VAR_PROTECTED` as necessary)

Reply:
Similarly there should perhaps be a `/RemoveResource(type, amount)`-proc which returns the amount actually acquired and, if all resources have been depleted, makes the extension remove itself from the host.

Reply:
Proc overhead for this would increase init time significantly, ore maps are generated repeatedly in worldgen.